src/logic/replay.h

```diff
@@ -74,6 +74,21 @@ class ReplayWriter {
 	StreamWrite* cmdlog_;
 	std::string filename_;
 };
+
+/** Extract the savegame from a replay to a temporary file. */
+struct ReplayPreloader {
```

Comment:
This should be called something like `ReplayfileSavegameExtractor` (yes, it needs improvement :) to reflect what it actually does. AFAICT it's not even used when the file is opened for replay viewing. (I think I can see the complications why it would be hard to unify.)